students/roviram/lesson07/html_render.py

```diff
@@ -0,0 +1,301 @@
+    """
+    Framework for 'p' paragraph tag, inherits from Element.
+    Overrides tag class attribute.
+    """
+    tag = 'p'
+
+
+class Head(Element):
+    """
+    Framework for 'head' tag, inherits from Element.
+    Overrides tag class attribute.
+    """
+    tag = 'head'
+
+
+class Li(Element):
+    """
+    Framework for 'li' list item tag, inherits from Element.
+    Overrides tag class attribute.
+    """
+    tag = 'li'
+
+
+class Ul(Element):
+    """
+    Framework for 'ul' unordered list tag, inherits from Element.
+    Overrides __init__ method and tag class attribute.
+    """
+    tag = 'ul'
+
+    def __init__(self, content=None, **kwargs):
+        """
+        __init__ method for Ul class.
+        :param content: Accepts no content, raises TypeError if content is passed it.
+        :param kwargs:  Used for passing in tag attributes.
+        """
+        if content: raise TypeError
+
+        self.contents = []
+        self.attributes = kwargs
+
+
+class OneLineTag(Element):
+    """
+    Framework for elements that render on a single line, inherits from Element.
+    Overrides append and render methods.
+    """
+    def append(self, new_content):
+        """
+        Raises NotImplementedError if called.
+        Overrides Element.append.
+        One line tags can not append.
+        """
+        raise NotImplementedError
+
+    def render(self, out_file, indent="", ind_count=0):
+        """
+        render method of OneLineTag class.
+        Renders tag and attributes on a single line
+        Overrides Element.render
+        :param out_file: Destination for rendered text.
+        :param indent: The specified indentation level for elements.
+        :param ind_count: How many times indent should be applied for the element.
+        """
+        self.indent = indent * ind_count
+
+        out_file.write(f'{self.indent}{self._open_tag()[:-1]}')
+        for key, value in self.attributes.items():
+            out_file.write(f' {key}="{value}"')
+        out_file.write(f'{self._open_tag()[-1:]}')
+        out_file.write(f'{self.contents[0]}')
+        out_file.write(f'{self._close_tag()}\n')
+
+
+class H(OneLineTag):
+    """
+    Framework for 'h' header tag, inherits from OneLineTag.
+    Overrides __init__ method and tag class attribute.
+    """
+    tag = 'h'
+
+    def __init__(self, level, content=None, **kwargs):
+        """
+        __init__ method for 'H' header class.
+        Calls super().__init__ after execution
+        :param content: Content that is to be added to the instance content and rendered.
+        :param kwargs:  Used for passing in tag attributes.
+        """
+
+        # Verifies function called with interger value for header level as first parameter.
+        # If it was forgotten, then contents will likely be the first passed value.
+        # This 'should' then fail if casted to an int, and raises a ValueError
+        try:
+            int(level)
+        except ValueError:
```

Comment:
This doesn't do much here right? If you catch the error, only to throw it again, there's nothing that's being done. We should throw a more meaningful error message or handle it in a graceful way.